File: src/main/scala/chapter01/Student.scala
package chapter01

class Student(name: String, var age: Int) {
    def printInfo(): Unit = {
        println(name + " " + age + " " + Student.school)
    }
}

// 引入伴生对象
object Student {
    val school: String = "hrbust"

    def main(args: Array[String]): Unit = {
        val tzk = new Student("tzk", 20)
        val jxd = new Student("jxd", 23)

        tzk.printInfo()
        jxd.printInfo()
    }
}

File: src/main/java/Student.java
public class Student {
    private String name;
    private Integer age;
    private static String school = "hrbust";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo(){
        System.out.println(this);
    }

    @Override
    public String toString() {
        return "{\"Student\":{"
                + "\"name\":\"" + name + "\""
                + ", \"age\":\"" + age + "\""
                + ", \"school\":\"" + Student.school + "\""
                + "}}";
    }

    public static void main(String[] args) {
        Student tzk = new Student("tzk", 23);
        Student jxd = new Student("jxd", 23);
        tzk.printInfo();
        jxd.printInfo();
    }
}
